geoalchemy2/__init__.py

```diff
@@ -105,10 +103,15 @@ def dispatch(event, table, bind):
                         stmt = stmt.execution_options(autocommit=True)
                         bind.execute(stmt)
                     elif bind.dialect.name == 'postgresql':
-                        bind.execute('CREATE INDEX "idx_%s_%s" ON "%s"."%s" '
-                                     'USING GIST ("%s")' %
-                                     (table.name, c.name, table_schema,
-                                      table.name, c.name))
+                        if table.schema:
+                            bind.execute('CREATE INDEX "idx_%s_%s" ON "%s"."%s" '
+                                         'USING GIST ("%s")' %
+                                         (table.name, c.name, table.schema,
+                                          table.name, c.name))
+                        else:
+                            bind.execute('CREATE INDEX "idx_%s_%s" ON "%s" '
+                                         'USING GIST ("%s")' %
+                                         (table.name, c.name, table.name, c.name))
```

Comment:
What about a helper function `table_ref(table)`  that returns the string? You could avoid the `if/else` and the duplication of SQL.

Reply:
At the same time the current code is very easy to read and understand. So I tend to think that the duplication is ok here.

Reply:
Loïc, I am going to merge this. But feel free to open a new PR if you think you can improve the code. I'll be happy to merge any improvement. Thanks.

Reply:
I was just passing by, it was a very superficial look at it.